components/form/ProjectForm.tsx

```diff
@@ -0,0 +1,187 @@
+import { Button, FormControl, FormLabel, Input, Select, FormErrorMessage } from '@chakra-ui/react';
+import { Box, Flex } from '@chakra-ui/layout';
+import React, { useState } from 'react';
+import { useRouter } from 'next/router';
+import * as fetch from '@/lib/utils/fetch';
+import ErrorAlert from '@/components/common/ErrorAlert';
+import { projectEndpointURL } from '@/lib/const';
+import { ProjectDTO, EmployeeDTO, CustomerDTO } from '@/lib/types/dto';
+import CustomerForm from './CustomerForm';
+
+export enum FormStatus {
+    IDLE = 'IDLE',
+    LOADING = 'LOADING',
+    ERROR = 'ERROR',
+    SUCCESS = 'SUCCESS',
+}
+
+const emptyProject: ProjectDTO = {
+    name: '',
+    description: undefined,
+    startDate: undefined,
+    endDate: undefined,
+    managingEmployee: undefined,
+    customer: undefined,
+    status: undefined,
+};
+
+type ProjectFormProps = {
+    employees: EmployeeDTO[];
+    customers: CustomerDTO[];
+    method: string;
+    project?: ProjectDTO;
+};
+
+function ProjectForm({ employees, customers, method, project: p }: ProjectFormProps): JSX.Element {
+    const [project, setProject] = useState<ProjectDTO>(p || emptyProject);
+    const [formStatus, setFormStatus] = useState<FormStatus>(FormStatus.IDLE);
+    const router = useRouter();
+
+    const handleCustomerChange = (e: React.FormEvent<HTMLSelectElement>) => {
+        e.preventDefault();
+        const id = e.currentTarget.value;
+        if (id) {
+            const customer = customers.find((el) => el.id === Number(id));
+            setProject({ ...project, customer: customer });
+        }
```

Comment:
This doesn't work, but it produces the same request as `main` branch. Need to fix.

Reply:
As a bonus, we reduce [clutter](https://github.com/three-consulting/epoc-frontend/pull/78/files#diff-0371e3443d5f2360efcc6f352cb9af86177163d97e5dcaf6fa718addc441eb26L63-L69)

Reply:
Fixed: This was a bug due to incorrect API types